src/api/flaskr/service/learn/preview_service.py

```
@@ -0,0 +1,415 @@
+            [prefer_draft, not prefer_draft]
+            if prefer_draft in (True, False)
+            else [True, False]
+        )
+        # ensure unique boolean list
+        struct_modes = list(dict.fromkeys(struct_modes))
+
+        for is_preview in struct_modes:
+            try:
+                struct = get_shifu_struct(self.app, shifu_bid, is_preview)
+            except Exception:
+                continue
```

Ruff (0.14.4)

`try`-`except`-`continue` detected, consider logging the exception

(S112)